src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        buffered_polygon = self.polygon.buffer(amount)
+
+        if isinstance(buffered_polygon, shapely.geometry.Polygon):
+            buffered_polygon = MultiPolygon([buffered_polygon])
+
+        return PolygonalFootprintRegion(polygon=buffered_polygon, name=self.name)
+
+    @cached_property
+    def isConvex(self):
+        return self.polygons.equals(self.polygons.convex_hull)
+
+class PathRegion(Region):
+    """A region composed of multiple polylines in 3D space.
+
+    One of points or polylines should be provided. If both are provided,
+    points will be used.
+
+    Args:
+        points: A list of points defining a single polyline.
+        polylines: A list of list of points, defining multiple polylines.
+    """
+    def __init__(self, points=None, polylines=None, name=None):
+        super().__init__(name)
+        # Standardize inputs
+        if points is not None:
+            polylines = [points]
+        elif polylines is not None:
+            pass
+        else:
+            raise ValueError("PathRegion created without specifying points or polylines")
+
+        # Extract a list of vertices and edges, without duplicates
+        self.vec_to_vert = {}
+        self.edges = []
+        vertex_iter = 0
+
+        for pls in polylines:
+            last_pt = None
+
+            for pt in pls:
+                # Extract vertex
+                cast_pt = toVector(pt)
+
+                if cast_pt not in self.vec_to_vert:
+                    self.vec_to_vert[cast_pt] = vertex_iter
+                    vertex_iter += 1
+
+                # Extract edge, if not first point
+                if last_pt is not None:
+                    new_edge = self.vec_to_vert[last_pt], self.vec_to_vert[cast_pt]
+                    self.edges.append(new_edge)
+
+                # Update last point
+                last_pt = cast_pt
+
+        self.vert_to_vec = {val:key for key,val in self.vec_to_vert.items()}
+        self.vertices = list(sorted(self.vec_to_vert.keys(), key=lambda vec: self.vec_to_vert[vec]))
+
+        # Extract length of each edge
+        self.edge_lengths = []
+
+        for edge in self.edges:
+            v1, v2 = edge
+            c1, c2 = self.vert_to_vec[v1], self.vert_to_vec[v2]
+
+            self.edge_lengths.append(c1.distanceTo(c2))
+
+    @lru_cache(maxsize=None)
+    def containsPoint(self, point, epsilon=1e-8):
+        pt = toVector(point)
+
+        # Check if each edge contains the point
+        for edge_iter, edge in enumerate(self.edges):
+            v1, v2 = edge
+            c1, c2 = self.vert_to_vec[v1], self.vert_to_vec[v2]
+
+            if (c1.distanceTo(pt) + pt.distanceTo(c2)) <= c1.distanceTo(c2) + epsilon:
+                return True
+
+        return False
+
+    def containsObject(self, obj):
+        return False
+
+    def containsRegionInner(self, reg, tolerance):
+        raise NotImplementedError
+
+    def distanceTo(self, point):
+        raise NotImplementedError
```

Comment:
Should be able to do this. If there's nothing in `trimesh` or one of our other dependencies that already does it, you could restore the `distanceToSegment` function that used to be in the `geometry` module (we removed it when we switched to Shapely, I think).